lambda/translmod.ml

```diff
@@ -1088,9 +1182,13 @@ let toploop_getvalue id =
   }
 
 let toploop_setvalue id lam =
+  if !Clflags.native_code then
+    (* See comment under [toploop_getvalue] *)
+    fatal_error "Translmod.toploop_setvalue: expected bytecode";
   Lapply{
     ap_loc=Loc_unknown;
-    ap_func=Lprim(mod_field toploop_setvalue_pos,
+    ap_func=Lprim(mod_field toploop_setvalue_pos
+                    (Module_value_only { field_count = -1 }),
```

Comment:
I don't love the `-1` here. It would be totally reasonable for someone to add a check in `mod_field` that the pos is compatible with the size of the module, for example. Maybe better to just inline this case of `mod_field`?